spaceinvaders.py

```diff
@@ -531,81 +527,53 @@ def create_main_menu(self):
-                                          score)
-                    self.explosionsGroup.add(explosion)
-                    self.allSprites.remove(currentSprite)
-                    self.enemies.remove(currentSprite)
-                    self.gameTimer = time.get_ticks()
-                    break
-
-        mysterydict = sprite.groupcollide(self.bullets, self.mysteryGroup,
-                                          True, True)
-        if mysterydict:
-            for value in mysterydict.values():
-                for currentSprite in value:
-                    currentSprite.mysteryEntered.stop()
-                    self.sounds['mysterykilled'].play()
-                    score = self.calculate_score(currentSprite.row)
-                    explosion = Explosion(currentSprite.rect.x,
-                                          currentSprite.rect.y,
-                                          currentSprite.row, False, True,
-                                          score)
-                    self.explosionsGroup.add(explosion)
-                    self.allSprites.remove(currentSprite)
-                    self.mysteryGroup.remove(currentSprite)
-                    newShip = Mystery()
-                    self.allSprites.add(newShip)
-                    self.mysteryGroup.add(newShip)
-                    break
-
-        bulletsdict = sprite.groupcollide(self.enemyBullets, self.playerGroup,
-                                          True, False)
-        if bulletsdict:
-            for value in bulletsdict.values():
-                for playerShip in value:
-                    if self.lives == 3:
-                        self.lives -= 1
-                        self.livesGroup.remove(self.life3)
-                        self.allSprites.remove(self.life3)
-                    elif self.lives == 2:
-                        self.lives -= 1
-                        self.livesGroup.remove(self.life2)
-                        self.allSprites.remove(self.life2)
-                    elif self.lives == 1:
-                        self.lives -= 1
-                        self.livesGroup.remove(self.life1)
-                        self.allSprites.remove(self.life1)
-                    elif self.lives == 0:
-                        self.gameOver = True
-                        self.startGame = False
-                    self.sounds['shipexplosion'].play()
-                    explosion = Explosion(playerShip.rect.x, playerShip.rect.y,
-                                          0, True, False, 0)
-                    self.explosionsGroup.add(explosion)
-                    self.allSprites.remove(playerShip)
-                    self.playerGroup.remove(playerShip)
-                    self.makeNewShip = True
-                    self.shipTimer = time.get_ticks()
-                    self.shipAlive = False
+        sprite.groupcollide(self.bullets, self.enemyBullets, True, True)
```

Comment:
Is this needed? Right below it we're looping over this, but it doesn't appear this line is doing anything.